src/storage/lock_manager/deadlock.rs

```diff
@@ -228,7 +255,8 @@ impl Inner {
             store_id,
             leader_info: None,
             leader_client: None,
-            detect_table: Rc::new(RefCell::new(DetectTable::default())),
+            // TODO: make it configurable.
+            detect_table: Rc::new(RefCell::new(DetectTable::new(time::Duration::from_secs(3)))),
```

Comment:
Does a valid pessimistic lock never last longer than 3 seconds?

Reply:
No. When a transaction meets a lock, it will send a `Detect` request. If the wait timeouts, it will send a `CleanUpWaitFor` request. The timeout of wait is configured by `wait-for-lock-timeout`.  
I will file a PR to use the `wait-for-lock-timeout` config as the `ttl`, so when wait timeouts, it won't send `CleanUpWaitFor` requests to the detector.

Reply:
You can think the max lifetime of a `WaitForEntry` is `wait-for-lock-timeout`. A new `Detect` request will extend its lifetime.

Reply:
I'm afraid if the transaction lasts longer than the timeout, things might go wrong.

Reply:
Oh, sorry. It should work well. LGTM.